bot/exts/info/python_news.py

```diff
@@ -216,16 +201,21 @@ async def post_maillist_news(self) -> None:
                     avatar_url=AVATAR_URL,
                     wait=True,
                 )
-                payload["data"][maillist].append(thread_information["thread_id"])
-
-                # Increase this specific maillist counter in stats
-                self.bot.stats.incr(f"python_news.posted.{maillist.replace('-', '_')}")
+                try:
+                    await self.bot.api_client.post(f"bot/mailing-lists/{maillist}/seen-items", json=thread_id)
+                    self.seen_items[maillist].add(thread_id)
+                    # Increase this specific maillist counter in stats
+                    self.bot.stats.incr(f"python_news.posted.{maillist.replace('-', '_')}")
 
-                if msg.channel.is_news():
-                    log.trace("Publishing mailing list message because it was in a news channel")
-                    await msg.publish()
+                    if msg.channel.is_news():
+                        log.trace("Publishing mailing list message because it was in a news channel")
+                        await msg.publish()
+                except ResponseCodeError as e:
+                    non_field_errors = e.response_json.get("non_field_errors", [])
+                    if non_field_errors == ["Seen item already known."]:
```

Comment:
What does this event cover? Inconsistent bot data?

Reply:
Yes, I'm not sure it'll hit that point but it's there for safety.
I ran into the issue when debugging, but I think it only happens because the loop is paused in the breakpoints.